### PR TITLE
Spring cleanup

### PR DESCRIPTION
**Pull request by dictoon**

### CODE REVIEW

File: src/renderqueueworker.cpp [hunk 1]
```diff
@@ -470,11 +99,9 @@ namespace
             if (renderingStarted && getWorldPtr()->getRenderState() != World::RSTATERENDERING)
                 break;
 
-            if (escPressed && (getWorldPtr()->getRenderState() == World::RSTATERENDERING))
+            if (GetAsyncKeyState(VK_ESCAPE) && getWorldPtr()->getRenderState() == World::RSTATERENDERING)
```
Comment (haggi) [inline src/renderqueueworker.cpp]: Does this work on linux?

File: src/renderqueueworker.cpp [hunk 2]
```diff
@@ -482,19 +109,25 @@ namespace
...
+                while (modifiedElementList.empty() &&
+                       getWorldPtr()->getRenderType() == World::IPRRENDER &&
+                       getWorldPtr()->getRenderState() != World::RSTATESTOPPED)
                     foundation::sleep(100);
-                if ((getWorldPtr()->getRenderType() != World::IPRRENDER) || (getWorldPtr()->getRenderState() == World::RSTATESTOPPED))
+
+                if (getWorldPtr()->getRenderType() != World::IPRRENDER ||
+                    getWorldPtr()->getRenderState() == World::RSTATESTOPPED)
                     break;
+
```
Comment (haggi) [inline src/renderqueueworker.cpp]: I think the whole renderThreadProc is a bit ugly but I did not yet find a way to make it better. Maybe with a condition variable to get rid of the sleep call?

File: src/renderqueueworker.cpp (hunk 1 above)
Comment (est77) [inline src/renderqueueworker.cpp]: Most likely not

**Comment by haggi** [comment]
I merged your pull request into my current branch. Now the renderview does often not update correctly any more. I had this behaviour before but it is quite some time ago. If I remember correctly it had something to to with the threads which trigger the updates. Did you change something in the way how the tilecallback is called?

**Comment by dictoon** [comment]
I'll have a look this week-end, I'll ping you on Hangout if you have a minute.

**Comment by haggi** [comment]
Just discovered another problem with the current workflow. At the moment the addIprCallbacks() are called from the renderthread what results in non working callbacks. Even if the nodeDirty callbacks work, the timerCallbacks seems not to like this process what means they simply do not work.

**Comment by dictoon** [comment]
> At the moment the addIprCallbacks() are called from the renderthread what results in non working callbacks.

I see. Let me revert my last "Spring cleanup" commit which was a bit too ambitious.

**Comment by dictoon** [comment]
Could you test again and let me know if that works better?

**Comment by haggi** [comment]
Okay, the IPR callbacks now work again. But the renderview does not update correctly - what means in IPR mode it does not update at all. I suppose we should put the renderview updates back to the renderqeueue.

**Comment by dictoon** [comment]
As discussed offline, I'll merge that PR, and then we can fix the Render View updates.

**Pull request merged**